lib/model.js

```diff
@@ -3106,6 +3108,10 @@ Model.$__insertMany = function(arr, options, callback) {
           _setIsNew(attribute, false);
         }
 
+        if (ordered === false && throwOnValidationError && validationErrors.length > 0) {
+          return callback(new Error('throwOnValidationError'), null);
```

Comment:
A good start, but I think we need a new error class for this. Can you add a `MongooseBulkWriteError` class that contains `response` from server, `validationErrors` list of validation errors, and `results` array?